src/main/daml/Daml/Finance/Instrument/Swap/Fpml/Util.daml

```diff
@@ -289,6 +289,20 @@ calculateFixPaymentClaimsFromSwapStream fixedRateSchedule s periodicSchedule cal
     assertMsg "Compounding not supported for the fixed rate leg" $
       isNone s.calculationPeriodAmount.calculation.compoundingMethodEnum
     let
+      initialStep = Step with
+        stepDate = (.unadjustedStartDate) $ head calculationSchedule
+        stepValue = fixedRateSchedule.initialValue
+      rateStepScheduleInclInitial = initialStep :: fixedRateSchedule.step
+
+      -- Find the fixed rate for the current date in the step list
```

Comment:
Can we just 
- sort the `step` array once and use `find` from the Prelude to find the correct date?

Or are these cases where `stepDate` does not match `c.unadjustedStartDate`?

Also, is there a relationship to be considered between
- `stepDate` in the coupon
- whether the rate is calculated in arrears or not ?

Also,

Reply:
> Or are these cases where `stepDate` does not match `c.unadjustedStartDate`?

Yes, this is the case. The only thing we know is that a step becomes effective on the `stepDate` onwards:
https://www.fpml.org/spec/fpml-5-11-3-lcwd-1/html/confirmation/schemaDocumentation/schemas/fpml-shared-5-11_xsd/complexTypes/Schedule/step.html

Reply:
> Also, is there a relationship to be considered between
> 
> * `stepDate` in the coupon
> * whether the rate is calculated in arrears or not ?

Good point. I think this is not a problem, because in arrears probably only refers to the floating leg (described by `CalculationPeriodEndDate` in the `resetDates` FpML Element), whereas this `FixedRateSchedule` only refers tot he fixed leg.

Reply:
Thanks for clarifying!

I think it's probably more efficient to reverse the `step` array in descending order once at the beginning.

Then you can use `find` from the prelude for the first item matching ` <= unadjustedStartDate`.

Reply:
I have changed this now. I also improved the step search for notionals in a similar way.

I thought about writing a generic function for it, but the code is very short and it also involves some data prep outside, so not sure much is gained from it. If there are more step searches later on, I will consider it again.